### PR TITLE
[EGGO-27] Implement directory structure from spec.

### PR DESCRIPTION
**Pull request by tomwhite**
This builds on https://github.com/bigdatagenomics/eggo/pull/22 to implement https://github.com/bigdatagenomics/eggo/issues/27. Basic and Flat editions have been implemented for both VCF and BAM.

Note that the BAM/SAM support relies on https://github.com/bigdatagenomics/adam/pull/650, but I think it may not have been working properly in Eggo before.

### CODE REVIEW

**Comment by laserson** [comment]
What's the rationale behind b2c650d? Isn't the idea that if something craps out midway through, you can restart the job and it'll only dnload what it needs?

Also, does this need to be rebased?

**Comment by tomwhite** [comment]
As it now stands (with the streaming job) it will only download everything in one go, but you're right - it would be good to have an incremental download. I need to change the streaming job to support that. I'll rebase this without that commit.

**Comment by tomwhite** [comment]
This is now rebased on master and doesn't have the incremental download change.

File: eggo/dag.py [hunk 1]
```diff
@@ -226,33 +242,38 @@ def run(self):
...
 
-class VCF2ADAMTask(Task):
+class ADAMBasicTask(Task):
 
     config = ConfigParameter()
+    adam_command = Parameter()
+    allowed_file_formats = Parameter()
+    edition = 'basic'
 
     def _raw_data_s3_url(self):
```
Comment (laserson) [inline eggo/dag.py]: can we get rid of these methods entirely and just use the module-level functions?

File: eggo/dag.py [hunk 2]
```diff
@@ -226,33 +242,38 @@ def run(self):
         p = Popen(delete_raw_cmd, shell=True)
         p.wait()
 
-class VCF2ADAMTask(Task):
+class ADAMBasicTask(Task):
```
Comment (laserson) [inline eggo/dag.py:243]: nice

File: eggo/dag.py [hunk 3]
```diff
@@ -273,46 +295,34 @@ def run(self):
     def output(self):
         return S3FlagTarget(self._target_s3_url())
 
-class BAM2ADAMTask(Task):
 
-    config = ConfigParameter()
+class ADAMFlatTask(Task):
```
Comment (laserson) [inline eggo/dag.py]: I like `ADAMFlattenTask` better, no?

**Comment by laserson** [comment]
Fixes #27

**Pull request merged**